app/javascript/react/components/ProductShowContainer.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react'
+import ProductShow from "./ProductShow"
+
+const ProductShowContainer = (props) => {
+  let productId = props.match.params.id
+
+  const [getProduct, setProduct] = useState({
+    id: null,
+    name: "",
+    description: "",
+    url: "",
+    image_url: ""
+  })
```

Comment:
lets swap lines 7-13 with line 5 to keep any `useState` statements at the top of the component